app/mod_auth/models.py

```diff
@@ -87,11 +87,29 @@ def request(self):
         db.session.add(self)
         db.session.commit()
 
-    def grant(self):
-        self.granted = True
+    def grant(self,approve):
```

Comment:
Here you can assign a default value to approve, such as True. This makes the definition more expressive as 'grant' implies that access is granted. When we would like to deny access, we can do so by explicitly passing False as an argument.